slatedb/src/compactor.rs

```diff
@@ -591,6 +619,28 @@ impl CompactorEventHandler {
         }
     }
 
+    /// Persists the current compactions state to the compactions store and refreshes the
+    /// local dirty object with the latest version.
+    async fn write_compactions_safely(&mut self) -> Result<(), SlateDBError> {
+        let desired_value = self.state.compactions_dirty().value.clone();
+        loop {
+            let mut dirty_compactions = self.compactions.prepare_dirty()?;
+            dirty_compactions.value = desired_value.clone();
+            match self.compactions.update(dirty_compactions).await {
+                Ok(()) => {
+                    let refreshed = self.compactions.prepare_dirty()?;
+                    self.state.set_compactions(refreshed);
+                    return Ok(());
+                }
+                Err(SlateDBError::TransactionalObjectVersionExists) => {
+                    // Retry with a refreshed view. Refresh will surface fencing if the epoch advanced.
```

Comment:
Just to make sure I got this (a comment would be good) a `TransactionalObjectVersionExists` can happen without fencing if something like an admin tool wrote a new version with a desired compaction?